examples/with-cookie-auth/pages/profile.js

```diff
@@ -41,28 +42,29 @@ const Profile = props => {
 
 Profile.getInitialProps = async ctx => {
   const { token } = nextCookie(ctx)
-  const url = `${process.env.API_URL}/api/profile.js`
+  const apiUrl = '/api/profile'
```

Comment:
This is not going to work for SSR, you can add this util instead:
```js
// This is not production ready, for that consider the usage of environment variables and NODE_ENV
function getHost(req) {
  if (!req) return '';

  const { host } = req.headers;

  if (host.startsWith('localhost')) {
    return `http://${host}`;
  }
  return `https://${host}`;
}
```
Then use it like so:
```js
const apiUrl = getHost(ctx.req) + '/api/profile'
```

Reply:
pretty cool, I remember we already discussed this 🤔. I will implement the function.